mod/common/src/main/java/gjum/minecraft/mapsync/common/utils/Hasher.java

```diff
@@ -0,0 +1,116 @@
+package gjum.minecraft.mapsync.common.utils;
+
+import io.netty.buffer.ByteBuf;
+import java.nio.ByteBuffer;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.Objects;
+import org.jetbrains.annotations.NotNull;
+
+public final class Hasher {
+	private final MessageDigest messageDigest;
+
+	private Hasher(
+			final @NotNull MessageDigest messageDigest
+	) {
+		this.messageDigest = Objects.requireNonNull(messageDigest);
+	}
+
+	/**
+	 * Updates the digest with a single byte.
+	 */
+	public @NotNull Hasher update(
+			final byte input
+	) {
+		this.messageDigest.update((byte) input);
+		return this;
+	}
+
+	/**
+	 * Updates the digest with an entire byte array.
+	 */
+	public @NotNull Hasher update(
+			final byte @NotNull [] input
+	) {
+		this.messageDigest.update((byte[]) input);
```

Comment:
Why are you casting the value here, and other times you input values into message digest?

Reply:
It's unnecessary, strictly speaking, but I found it made the code easier to read given how, without them, most of the methods look identical to one another, they start to blur together, and it becomes more difficult to grasp what's going on, a bit like when you do too much promise chaining in Javascript. I can remove the casts if need be, but I don't think they'll burden the JVM (I think they'll get optimised out?), and I do think they can be useful is helping distinguish between the different method bodies.